### ISSUE TITLE
Shop API should not use ThemeBundle to determine the right locale

### ISSUE THREAD

**Issue by CSchulz** [Sylius/ShopApiPlugin#561]
Calling `shop-api/taxons/` loads the configuration via ThemeBundle, this should be reconsidered to be able to skip the channel and use the default channel.

Example stacktrace:
```
Sylius\Component\Channel\Context\ChannelNotFoundException:
Channel could not be found! Tip: You can use the Web Debug Toolbar to switch between channels in development.

  at vendor/sylius/sylius/src/Sylius/Component/Channel/Context/CompositeChannelContext.php:47
  at Sylius\Component\Channel\Context\CompositeChannelContext->getChannel()
     (vendor/sylius/sylius/src/Sylius/Component/Channel/Context/CachedPerRequestChannelContext.php:59)
  at Sylius\Component\Channel\Context\CachedPerRequestChannelContext->getChannel()
     (vendor/sylius/sylius/src/Sylius/Bundle/CoreBundle/Theme/ChannelBasedThemeContext.php:44)
  at Sylius\Bundle\CoreBundle\Theme\ChannelBasedThemeContext->getTheme()
     (vendor/sylius/theme-bundle/src/Translation/ThemeAwareTranslator.php:114)
  at Sylius\Bundle\ThemeBundle\Translation\ThemeAwareTranslator->transformLocale('en_US')
     (vendor/sylius/theme-bundle/src/Translation/ThemeAwareTranslator.php:89)
  at Sylius\Bundle\ThemeBundle\Translation\ThemeAwareTranslator->setLocale('en_US')
     (vendor/symfony/http-kernel/EventListener/LocaleAwareListener.php:70)
  at Symfony\Component\HttpKernel\EventListener\LocaleAwareListener->setLocale('en_US', 'en_US')
     (vendor/symfony/http-kernel/EventListener/LocaleAwareListener.php:43)
  at Symfony\Component\HttpKernel\EventListener\LocaleAwareListener->onKernelRequest(object(RequestEvent), 'kernel.request', object(TraceableEventDispatcher))
     (vendor/symfony/event-dispatcher/Debug/WrappedListener.php:126)
  at Symfony\Component\EventDispatcher\Debug\WrappedListener->__invoke(object(RequestEvent), 'kernel.request', object(TraceableEventDispatcher))
     (vendor/symfony/event-dispatcher/EventDispatcher.php:260)
  at Symfony\Component\EventDispatcher\EventDispatcher->doDispatch(array(object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener)), 'kernel.request', object(RequestEvent))
     (vendor/symfony/event-dispatcher/EventDispatcher.php:235)
  at Symfony\Component\EventDispatcher\EventDispatcher->callListeners(array(object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener), object(WrappedListener)), 'kernel.request', object(RequestEvent))
     (vendor/symfony/event-dispatcher/EventDispatcher.php:73)
  at Symfony\Component\EventDispatcher\EventDispatcher->dispatch(object(RequestEvent), 'kernel.request')
     (vendor/symfony/event-dispatcher/Debug/TraceableEventDispatcher.php:168)
  at Symfony\Component\EventDispatcher\Debug\TraceableEventDispatcher->dispatch(object(RequestEvent), 'kernel.request')
     (vendor/symfony/http-kernel/HttpKernel.php:127)
  at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Request), 1)
     (vendor/symfony/http-kernel/HttpKernel.php:68)
  at Symfony\Component\HttpKernel\HttpKernel->handle(object(Request), 1, true)
     (vendor/symfony/http-kernel/Kernel.php:198)
  at Symfony\Component\HttpKernel\Kernel->handle(object(Request))
     (public/index.php:25)
```

**Comment by mamazu** [comment]
**TLDR:** It's a Sylius problem and has nothing to do with ShopAPI and is not a problem when resolving the locale but what happens afterwards.

## The long explanation of what you see
As part of my pull request of adding the locale from a header I dug into how Sylius handles locales. In general Symfony determines the locale of the request when "accepting" the request and fires a "kernel.request" event. Many processes hook into this event, like the `RequestLocaleSetter` which has `LocaleContext` to resolve the locale. Everything fine up to this point. The locale is correctly resolved as it uses the Symfony fallback locale.

However before this all can happen Symfony is trying to resolve the locale. (Symfony's service has priority 16 and 15 and Sylius' Services have 4 and 3 so that is an issue but not the one your are witnessing.) And after Sylius is done resolving the locale [there is a service](https://github.com/symfony/symfony/blob/14080ce5b5dbd3a7e469577bf781dd23bf46870e/src/Symfony/Component/HttpKernel/EventListener/LocaleAwareListener.php#L68) `Symfony\Component\HttpKernel\EventListener\LocaleAwareListener` that calls the `setLocale` function on every service that is tagged as locale aware.

*Enter your stacktrace*

One of the services that is trying to set the locale on is the `ThemeBundle`. It needs that for making the locale theme specific. And how does it get the theme? Well the theme is tied to a channel (or to be more precise the channel has the theme). So it tries to get the channel and there it fails because there is no channel in your application.
